AdyenDropIn/Modules/Component/ComponentContainerRouter.swift

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2025 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import Foundation
+
+internal protocol ComponentContainerRouterDelegate: AnyObject {
+    func didSubmit(_ data: PaymentComponentData, from component: any PaymentComponent)
+    func didFail(with error: any Error)
+    func didCancel(component: any PaymentComponent)
+}
+
+internal protocol ComponentContainerRouterProtocol {
+    var delegate: ComponentContainerRouterDelegate? { get set }
+    var rootViewController: UIViewController { get }
+    func start()
+}
+
+internal class ComponentContainerRouter: ComponentContainerRouterProtocol {
+
+    // MARK: - Properties
+
+    internal weak var delegate: ComponentContainerRouterDelegate?
+    private let view: UIViewController
+    private var viewModel: ComponentContainerViewModelProtocol
+
+    // MARK: - Initializers
+
+    internal init(
```

Comment:
Wou can also pass the delegate in the init if there is no scenario where delegate should not be set.

Reply:
Done!